dds/idl/typeobject_generator.cpp

```diff
@@ -225,28 +225,58 @@ typeobject_generator::gen_struct(AST_Structure*, UTL_ScopedName* name,
     const string decl_gto = "getMinimalTypeObject<" + clazz + ">";
     Function gto(decl_gto.c_str(), "const XTypes::TypeObject&", "");
     gto.endArgs();
-
+    const ExtensibilityKind exten = be_global->extensibility(node);
+    string bitwise_str;
+    switch (exten) {
+      case extensibilitykind_final:
+        bitwise_str = "XTypes::IS_FINAL";
+        break;
+      case extensibilitykind_appendable:
+        bitwise_str = "XTypes::IS_APPENDABLE";
+        break;
+      case extensibilitykind_mutable:
+        bitwise_str = "XTypes::IS_MUTABLE";
+        break;
+      default:
+        idl_global->err()->misc_error(
+          "Unexpected extensibility while setting flags", node);
+        return false;
+    }
+    if (!be_global->is_topic_type(node)) {
+      bitwise_str += " | XTypes::IS_NESTED";
+    }
+    if (0) { //TODO: Change once HASHID is supported in be_global.cpp
+      bitwise_str += " | XTypes::IS_AUTOID_HASH";
+    }
     // TODO: Support struct inheritance.
     be_global->impl_ <<
       "  static const XTypes::TypeObject to = XTypes::TypeObject(\n"
       "    XTypes::MinimalTypeObject(\n"
       "      XTypes::MinimalStructType(\n"
-      "        XTypes::IS_FINAL | XTypes::IS_NESTED | XTypes::IS_AUTOID_HASH,\n" // TODO: Pick the appropriate flags.
+      "        " << bitwise_str << ",\n"
       "        XTypes::MinimalStructHeader(\n"
       "          getMinimalTypeIdentifier<void>(),\n"
       "          XTypes::MinimalTypeDetail()\n"
       "        ),\n"
       "        XTypes::MinimalStructMemberSeq()\n";
 
     ACE_CDR::ULong member_id = 0;
+
     for (std::vector<AST_Field*>::const_iterator pos = fields.begin(), limit = fields.end(); pos != limit; ++pos) {
+      string member_string;
+      bool value;
+      if (be_global->check_key(*pos, value)) {
```

Comment:
I missed this before:

```suggestion
      bool is_key;
      be_global->check_key(*pos, is_key);
      if (is_key) {
```

The return value is just if the key annotation is there or not. The key annotation itself can be `@key`, `@key(TRUE)` (which both mean the same thing), or `@key(FALSE)`.

Reply:
This ends up returning node_value_exists which the comment says 

 /**
   * If node has the annotation, this sets value to the annotation value and
   * returns true.
   * If node does not have the annotation, this sets value to default_value()
   * and returns false.
   */

so I don't think this is a problem.  Please let me know if I am misunderstanding this.